website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,223 @@
+        messages,
+        max_tokens: variant.max_tokens,
+        temperature: variant.temperature,
+    });
+    return result.toDataStreamResponse();
+};
+```
+
+This file is doing a few key things:
+
+1. Sets up our OpenAI client with an API key.
+2. Defines different AI model configurations in the `variants` object.
+3. Handles incoming chat requests with the `POST` function.
+
+When a request comes in, it:
+
+-   Extracts the messages from the request body.
+-   Selects a variant (currently hardcoded to 'variant1').
+-   Uses the OpenAI API to generate a response.
+-   Streams the response back to the client.
+
+The `streamText` function is one of the utilities provided by Vercel's AI SDK. It helps deal with the real-time streaming of the AI's responses.
+
+## Creating a feature flag with AI variants
+
+Instead of hardcoding `variant1`, we want to use feature flags to dynamically choose which AI model to use. This will let us easily switch between models, test different configurations, or even do some A/B testing to see which model performs better for which task.
+
+To implement this, we'll need to:
+
+1. Set up a feature flag provider (we'll use Unleash).
+2. Replace our static variant selection with feature flag calls.
+3. Use the feature flag in our code to determine which AI model and settings to use for each request.
+
+### Install a local feature flag provider
+
+In this section, we’ll install Unleash, run the instance locally, log in, and create a feature flag. If you prefer, you can use other tools instead of Unleash, but you’ll need to update the code accordingly.
+
+Use Git to clone the Unleash repository and Docker to build and run it. Open a terminal window and run the following commands:
+
+```sh
+git clone https://github.com/unleash/unleash.git
+cd unleash
+docker compose up -d
+```
+
+You now have Unleash installed on your machine and running in the background. You can access this instance in your web browser at [http://localhost:4242](http://localhost:4242).
+
+Log in to the platform using these credentials:
+
+```
+Username: admin
+Password: unleash4all
+```
+
+Click **New feature flag** to create a new feature flag.
+
+![Create a new feature flag](../ruby/new-ff.png)
+
+Create a new feature flag called `gpt-version`.
+
+After that, and this the most important part, we need to add a variant to our feature flag.
+
+What we'll do is hold all the model configurations in the feature flag variants.
+
+So, we can take these model configurations:
+
+```javascript
+const variants = {
+    variant1: {
+        model: "gpt-3.5-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant2: {
+        model: "gpt-4-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant3: {
+        model: "gpt-4-vision-preview",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+};
+```
+
+And add them into Unleash.
+
+![a variant with parameters for an OpenAI model](./model-variant.png)
```

Comment:
Normally we take Admin UI screenshots in light mode. I think it's ok for it to be dark mode, if it's the only Admin UI screenshot on the page, or if all of them are consistent. Having some light and some dark might easily give the impression that it's not the same interface. Regardless of the share, could you please use one of these? Adjusted the spacing a bit and tried to not cut off UI elements as much as possible

<img width="1106" alt="Screenshot 2024-10-17 at 09 36 12" src="https://github.com/user-attachments/assets/776ed67d-7406-48eb-8fb3-eadd329091b2">
<img width="1106" alt="Screenshot 2024-10-17 at 09 34 55" src="https://github.com/user-attachments/assets/d46b17a6-c726-46bb-958a-3f2328c6aef5">